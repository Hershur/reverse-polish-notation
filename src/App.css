#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media (prefers-reduced-motion: no-preference) {
  a:nth-of-type(2) .logo {
    animation: logo-spin infinite 20s linear;
  }
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}


/*
Basic input element using psuedo classes
*/

html {
  font-family: 'Lora', sans-serif;
  width: 100%;
}

body {
  margin: 5% auto 0 auto;
  width: 90%;
  max-width: 1125px;
}

h1 {
  font-size: 28px;
  margin-bottom: 2.5%;
}

input,
span,
label,
textarea {
  font-family: 'Ubuntu', sans-serif;
  display: block;
  margin: 10px;
  padding: 5px;
  border: none;
  font-size: 22px;
}

textarea:focus,
input:focus {
  outline: 0;
}
/* Question */

input.rpn-input,
textarea.rpn-input {
  font-size: 48px;
  font-weight: 300;
  border-radius: 2px;
  margin: 0;
  border: none;
  width: 80%;
  background: rgba(0, 0, 0, 0);
  transition: padding-top 0.2s ease, margin-top 0.2s ease;
  overflow-x: hidden; /* Hack to make "rows" attribute apply in Firefox. */
}
/* Underline and Placeholder */

input.rpn-input + label,
textarea.rpn-input + label {
  display: block;
  position: relative;
  white-space: nowrap;
  padding: 0;
  margin: 0;
  width: 10%;
  border-top: 1px solid red;
  -webkit-transition: width 0.4s ease;
  transition: width 0.4s ease;
  height: 0px;
}

input.rpn-input:focus + label,
textarea.rpn-input:focus + label {
  width: 80%;
}

input.rpn-input:focus,
input.rpn-input:valid {
  padding-top: 35px;
}

textarea.rpn-input:valid,
textarea.rpn-input:focus {
  margin-top: 35px;
}

input.rpn-input:focus + label > span,
input.rpn-input:valid + label > span {
  top: -100px;
  font-size: 22px;
  color: #333;
}

textarea.rpn-input:focus + label > span,
textarea.rpn-input:valid + label > span {
  top: -150px;
  font-size: 22px;
  color: #333;
}

input.rpn-input:valid + label,
textarea.rpn-input:valid + label {
  border-color: green;
}

input.rpn-input:invalid,
textarea.rpn-input:invalid {
  box-shadow: none;
}

input.rpn-input + label > span,
textarea.rpn-input + label > span {
  font-weight: 300;
  margin: 0;
  position: absolute;
  color: #8F8F8F;
  font-size: 48px;
  top: -66px;
  left: 0px;
  z-index: -1;
  -webkit-transition: top 0.2s ease, font-size 0.2s ease, color 0.2s ease;
  transition: top 0.2s ease, font-size 0.2s ease, color 0.2s ease;
}

input[type="submit"] {
  -webkit-transition: opacity 0.2s ease, background 0.2s ease;
  transition: opacity 0.2s ease, background 0.2s ease;
  display: block;
  opacity: 0;
  margin: 10px 0 0 0;
  padding: 10px;
  cursor: pointer;
}

input[type="submit"]:hover {
  background: #EEE;
}

input[type="submit"]:active {
  background: #999;
}

input.rpn-input:valid ~ input[type="submit"], textarea.rpn-input:valid ~ input[type="submit"] {
  -webkit-animation: appear 1s forwards;
  animation: appear 1s forwards;
}

input.rpn-input:invalid ~ input[type="submit"], textarea.rpn-input:invalid ~ input[type="submit"] {
  display: none;
}

@-webkit-keyframes appear {
  100% {
    opacity: 1;
  }
}

@keyframes appear {
  100% {
    opacity: 1;
  }
}

.result-box {
  margin-top: 5rem;
  font-size: 2.5rem;
}
