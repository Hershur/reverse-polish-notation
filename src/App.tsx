import './App.css'
import RPNBox from './components/RPNBox'

function App() {

  return (
    <>
     <RPNBox />
    </>
  )
}

export default App
