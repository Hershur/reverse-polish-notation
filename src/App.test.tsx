import App from "./App";
import { render } from "@testing-library/react";


describe('Snapshot testing suite for whole App', () => {
    it('Matches DOM Snapshot', () => {
      const { asFragment } = render(<App />);
      expect(asFragment()).toMatchSnapshot();
    });
});