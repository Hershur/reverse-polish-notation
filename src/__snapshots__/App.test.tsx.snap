// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing suite for whole App Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="rpnBox"
  >
    <h1
      data-testid="heading"
    >
      Reverse Polish Notation Calculator
    </h1>
    <div>
      <input
        autocomplete="off"
        class="rpn-input"
        data-testid="rpnInput"
        id="rpn"
        name="rpn"
        required=""
        type="text"
        value=""
      />
      <label
        for="rpn"
      >
        <span>
          Enter reverse polish notation
        </span>
      </label>
    </div>
    <div
      class="result-box"
      data-testid="resultBox"
    />
  </div>
</DocumentFragment>
`;
